### PR TITLE
Suspend field should be mutable

### PR DESCRIPTION
**Pull request by mcariatm**
### What would you like to be added:
we reset StartTime to allow updating the nodeSelectors on the job. However StartTime is in status, which we can update via a status update call only

### Why is this needed:
Update start time using client.Status().Update(..)

### Issue Number:
#156 

/kind bug

### CODE REVIEW

File: test/integration/webhook/jobset_webhook_test.go [hunk 1]
```diff
@@ -270,5 +271,42 @@ var _ = ginkgo.Describe("jobset webhook defaulting", func() {
 			},
 			jobSetCreationShouldFail: true,
 		}),
+		ginkgo.Entry("jobSet should suspended", &testCase{
```
Comment (danielvegamyhre) [inline test/integration/webhook/jobset_webhook_test.go:274]:
```suggestion
		ginkgo.Entry("suspend jobset", &testCase{
```

File: test/integration/webhook/jobset_webhook_test.go [hunk 2]
```diff
@@ -270,5 +271,42 @@ var _ = ginkgo.Describe("jobset webhook defaulting", func() {
...
+			},
+			defaultsApplied: func(js *jobset.JobSet) bool {
+				return true
+			},
+			updateJobSet: func(js *jobset.JobSet) {
+				js.Spec.Suspend = pointer.Bool(true)
+			},
+			updateShouldFail: false,
+		}),
+		ginkgo.Entry("jobSet should unsuspended", &testCase{
```
Comment (danielvegamyhre) [inline test/integration/webhook/jobset_webhook_test.go:292]:
```suggestion
		ginkgo.Entry("resume jobset", &testCase{
```

**Review by danielvegamyhre — CHANGES_REQUESTED** [review]

File: test/integration/webhook/jobset_webhook_test.go [hunk 3]
```diff
@@ -270,5 +271,42 @@ var _ = ginkgo.Describe("jobset webhook defaulting", func() {
...
+				return testing.MakeJobSet("js-hostnames-non-indexed", ns.Name).
+					ReplicatedJob(testing.MakeReplicatedJob("rjob").
+						Job(testing.MakeJobTemplate("job", ns.Name).
+							PodSpec(testing.TestPodSpec).
+							CompletionMode(batchv1.IndexedCompletion).Obj()).
+						EnableDNSHostnames(true).
+						Obj())
+			},
+			defaultsApplied: func(js *jobset.JobSet) bool {
+				return true
```
Comment (danielvegamyhre) [inline test/integration/webhook/jobset_webhook_test.go:285]: No need to define this if it isn't validating defaults were applied. The `defaultsApplied` function is only executed if it is not nil. In fact, can we remove this from other test cases which do this as well?

File: api/v1alpha1/jobset_webhook.go [hunk 4]
```diff
@@ -96,8 +96,8 @@ func (js *JobSet) ValidateUpdate(old runtime.Object) error {
 		}
 	}
 
-	if !apiequality.Semantic.DeepEqual(oldSpec, *mungedSpec) {
-		specDiff := cmp.Diff(oldSpec, *mungedSpec)
+	if !apiequality.Semantic.DeepEqual(oldSpec.ReplicatedJobs, mungedSpec.ReplicatedJobs) {
+		specDiff := cmp.Diff(oldSpec.ReplicatedJobs, mungedSpec.ReplicatedJobs)
 		return field.Forbidden(field.NewPath("spec"), fmt.Sprintf("jobset updates may not change fields other than %s\n%v", strings.Join(updatableFields, ","), specDiff))
 	}
 	return nil
```
Comment (ahg-g) [inline api/v1alpha1/jobset_webhook.go:103]:
```suggestion
  allErrs = append(allErrs, apivalidation.ValidateImmutableField(mungedSpec.ReplicatedJobs, oldSpec.ReplicatedJobs, specPath.Child("replicatedJobs"))...)
  allErrs = append(allErrs, apivalidation.ValidateImmutableField(mungedSpec.SuccessPolicy, oldSpec.SuccessPolicy, specPath.Child("successPolicy"))...)
  allErrs = append(allErrs, apivalidation.ValidateImmutableField(mungedSpec.FailurePolicy, oldSpec.FailurePolicy, specPath.Child("failurePolicy"))...)
return allErrs   
```

Comment (ahg-g) [reply api/v1alpha1/jobset_webhook.go:103]: and import `apivalidation "k8s.io/apimachinery/pkg/api/validation"`

**Review by ahg-g — COMMENTED** [review]
success and failure policy fields should remain immutable, see the suggested fix

File: api/v1alpha1/jobset_webhook.go (hunk 4 above)
Comment (mcariatm) [reply api/v1alpha1/jobset_webhook.go:103]: they are immutable. See on jobset_types.go line 53 and 60

Comment (mcariatm) [reply api/v1alpha1/jobset_webhook.go:103]: I can add some tests to demonstrate that....

Comment (mcariatm) [reply api/v1alpha1/jobset_webhook.go:103]: Conclusion:
After some tests, FailurePolicy is immutable only if it was set before. Should I do something else on this side?

Comment (ahg-g) [reply api/v1alpha1/jobset_webhook.go:103]: ah, right, I updated the suggestion, lets still use `apivalidation.ValidateImmutableField` to check the replicatedJobs field and have a comment indicating that success and failure policy are made immutable via CEL

Comment (mcariatm) [reply api/v1alpha1/jobset_webhook.go]: done

**Pull request merged**